syntax = "proto3";

package dwarf;

// WebSocket-Paket
message WsPacket {
  uint32 major_version = 1;  // Protokoll-Hauptversion
  uint32 minor_version = 2;  // Protokoll-Nebenversion
  uint32 device_id = 3;      // Ger√§te-ID (1: DWARF II, 2: MAGNI)
  uint32 module_id = 4;      // Modul-ID
  uint32 cmd = 5;            // Befehl
  uint32 type = 6;           // 0: Request, 1: Response, 2: Notification, 3: Response
  bytes data = 7;            // Daten (serialisierte Message)
  string client_id = 8;      // WebSocket Client-ID
}

// Standard-Response
message ComResponse {
  int32 code = 1;  // Fehlercode (0 = Erfolg)
}
